File: 94.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */



var inorderTraversal = function (root) {

    let num = []

    function inorderTraversal2(root) {

        if (!root) return []
        if (!root.left) {
            num.push(root.val)
            inorderTraversal2(root.right)
        }
        if (root.left) {
            inorderTraversal2(root.left)
            num.push(root.val)
            inorderTraversal2(root.right)
        }
        return num
    }

    return inorderTraversal2(root)

};

File: 94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.arr = []
    def re (self,root):
        if root is None:
            return
        self.inorderTraversal(root.left)
        self.arr.append(root.val)
        self.inorderTraversal(root.right)

    def inorderTraversal(self, A: Optional[TreeNode]) -> List[int]:
        self.re(A)
        return self.arr